app/controllers/providers/used_multiple_delegated_functions_controller.rb

```diff
@@ -0,0 +1,92 @@
+module Providers
+  class UsedMultipleDelegatedFunctionsController < ProviderBaseController
+    include PreDWPCheckVisible
+
+    def show
+      form
+    end
+
+    def update
+      render :show unless save_continue_and_update_scope_limitations
+    end
+
+    private
+
+    def save_continue_and_update_scope_limitations
+      form.draft = draft_selected?
+      return unless form.save(form_params)
+
+      update_application
+
+      draft_selected? ? continue_or_draft : go_forward(earliest_df&.used_delegated_functions_reported_on)
+    end
+
+    def form
+      @form ||= LegalAidApplications::UsedMultipleDelegatedFunctionsForm.call(application_proceedings_by_name)
+    end
+
+    def proceeding_types
+      @proceeding_types ||= legal_aid_application.proceeding_types
+    end
+
+    def application_proceedings_by_name
+      @application_proceedings_by_name ||= legal_aid_application.application_proceedings_by_name
+    end
+
+    def application_proceeding_types
+      application_proceedings_by_name.map(&:application_proceeding_type)
+    end
+
+    def earliest_df
+      form.earliest_delegated_functions
+    end
+
+    def earliest_df_date
+      earliest_df&.used_delegated_functions_on
+    end
```

Comment:
I wonder if naming could be better on those two methods. Maybe avoid using `df` instead of `delegated_functions`, so `earliest_df_date` becomes `earliest_delegated_functions_date`(longer but clearer).

Also it's not completely clear what `earliest_df` actually returns - maybe something like `proceeding_with_earliest_delegated_functions` would be clearer.

Reply:
👍